numpy/core/src/npymath/halffloat.c

```diff
@@ -307,9 +307,11 @@ npy_uint16 npy_floatbits_to_halfbits(npy_uint32 f)
         /*
          * If the last bit in the half significand is 0 (already even), and
          * the remaining bit pattern is 1000...0, then we do not add one
-         * to the bit after the half significand.  In all other cases, we do.
+         * to the bit after the half significand. However, since we may have
+         * previously lost up to 11 bits, we need to check these as well.
+         * In all other cases, we can just add one.
          */
-        if ((f_sig&0x00003fffu) != 0x00001000u) {
+        if (((f_sig&0x00003fffu) != 0x00001000u) || (f&0x008ffu)) {
```

Comment:
Where did `008ff` come from? And why are you giving only 5 hex digits?

Reply:
Jeesh, I messed up, probably a typo, it should have been `7ff` and of course full digits. 7ff for the last 11 bits, that would otherwise be gone.

Reply:
Dang, the tests probably do not catch this, because it is in the wrong bits... maybe should add a test for this part more specifically.